src/SAML2/XML/alg/DigestMethod.php

```diff
@@ -65,27 +56,47 @@ public function getAlgorithm(): string
      * @param string $algorithm
      * @return void
      */
-    public function setAlgorithm(string $algorithm): void
+    private function setAlgorithm(string $algorithm): void
     {
         $this->Algorithm = $algorithm;
     }
 
 
+    /**
+     * Convert XML into a DigestMethod
+     *
+     * @param \DOMElement $xml The XML element we should load
+     * @return self
+     */
+    public static function fromXML(DOMElement $xml): object
+    {
+        if (!$xml->hasAttribute('Algorithm')) {
+            throw new \Exception('Missing required attribute "Algorithm" in alg:DigestMethod element.');
+        }
+
+        return new self($xml->getAttribute('Algorithm'));
+    }
+
+
     /**
      * Convert this element to XML.
      *
-     * @param \DOMElement $parent The element we should append to.
+     * @param \DOMElement|null $parent The element we should append to.
      * @return \DOMElement
      *
      * @throws \InvalidArgumentException if assertions are false
      */
-    public function toXML(DOMElement $parent): DOMElement
+    public function toXML(DOMElement $parent = null): DOMElement
     {
-        Assert::notEmpty($this->Algorithm, 'Cannot convert DigestMethod to XML without an Algorithm set.');
+        if ($parent === null) {
```

Comment:
Maybe it's a good idea to have a very simple PR first with this code in a method inside `AbstractConvertable`, and then use it in these PRs instead of having to copy/paste the code?

Reply:
Sure, let's do that right after this one.. Unless you've spotted issues?

Reply:
Haven't finished yet, but not so far...

Reply:
The method was added, but it depends on #199 to be merged